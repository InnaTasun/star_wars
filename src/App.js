import Main from "./сomponents/Main";
import Movie from "./сomponents/Movie";
import styled from "styled-components";
import { BrowserRouter as Router, Route, Routes, Link } from "react-router-dom";
import homeLogo from "./images/homeLogo.png";
import Search from "./сomponents/Search";
import SearchResults from "./сomponents/SearchResults";
import Character from "./сomponents/Character";

function App(props) {
  return (
    <div className={`${props.className} content`}>
      <Router>
        <header className="content--header header">
          <Link to="/">
            <img className="header--home-link" src={homeLogo} alt="home logo" />
          </Link>
          <Search className="header--search" />
        </header>

        <Routes>
          <Route path="/" element={<Main />} />
          <Route path="/search/:q" element={<SearchResults />} />
          <Route path="/films/:id" element={<Movie />} />
          <Route path="/people/:id" element={<Character />} />
        </Routes>
      </Router>

      <footer className="content--footer">
        <p>Created by Inna Tasun &copy; 2021</p>
      </footer>
    </div>
  );
}

const StyledApp = styled(App)`
  min-height: 100vh;
  min-width: 700px;
  display: flex;
  flex-direction: column;
  justify-content: space-between;

  * {
    margin: 0;
    padding: 0;
    box-sizing: border-box;
    font-family: Helvetica, Verdana, Arial, sans-serif;
  }

  @media (max-width: 1130px) {
    aside {
      display: none;
    }
  }

  .content {
    width: 100%;
    padding: 1rem;

    &--header {
      height: 40px;
    }

    &--footer {
      text-align: center;
      vertical-align: bottom;
      margin-top: 40px;
    }
  }

  .header {
    display: flex;
    justify-content: space-between;
    align-items: flex-end;
    margin-bottom: 20px;

    &--home-link {
      height: 40px;
      margin-bottom: -5px;
    }

    &--search {
      height: 30px;
    }
  }
`;

export default StyledApp;
