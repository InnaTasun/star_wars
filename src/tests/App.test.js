import React from "react";
import { render, screen } from "@testing-library/react";
import App from "../сomponents/App";

describe("App component", () => {
  test("renders with data", () => {
    render(<App />);
    expect(screen.getByRole("searchbox")).toBeInTheDocument();
    expect(screen.getAllByRole("img").length).toBe(3);
    expect(screen.getByRole("heading")).toBeInTheDocument();
    expect(screen.getAllByRole("link").length).toBe(2);
    expect(screen.getByText(/created by/i)).toBeInTheDocument();
  });

  test("renders with or without className", () => {
    let container = render(<App className="main-class" />).container;
    let firstDiv = container.firstElementChild;
    expect(firstDiv.className).toContain("main-class");

    container = render(<App className="test-class" />).container;
    firstDiv = container.firstElementChild;
    expect(firstDiv.className).toContain("test-class");

    container = render(<App />).container;
    firstDiv = container.firstElementChild;
    expect(firstDiv.className).not.toContain("main-class");
  });
});

test("App snapshot", () => {
  const { container } = render(<App />);
  expect(container).toMatchSnapshot();
});
