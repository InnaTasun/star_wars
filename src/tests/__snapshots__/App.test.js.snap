// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App snapshot 1`] = `
<div>
  <div
    class="sc-jrQzAO jFjnHv content"
  >
    <header
      class="content--header header"
    >
      <a
        href="/"
      >
        <img
          alt="home logo"
          class="header--home-link"
          src="homeLogo.png"
        />
      </a>
      <form
        action="/search/"
        class="sc-gKclnd pXNNj header--search search"
      >
        <input
          class="search--input"
          placeholder="Search movie or character..."
          type="search"
          value=""
        />
        <a
          href="/search/"
        >
          <img
            alt="lens"
            class="search--btn"
            src="searchImg.png"
          />
        </a>
      </form>
    </header>
    <div
      class="sc-dkPtRN iYKAki"
    >
      <div
        class="main-logo"
      >
        <img
          alt="main logo"
          class="main-logo--img"
          src="mainLogo.jpeg"
        />
      </div>
      <h3>
        MOVIES
      </h3>
      <ul
        class="sc-gsDKAQ enRQhp films-list"
      />
      <br />
    </div>
    <footer
      class="content--footer"
    >
      <p>
        Created by Inna Tasun Â© 2021
      </p>
    </footer>
  </div>
</div>
`;
